evap/evaluation/views.py

```diff
@@ -75,6 +76,12 @@ def index(request):
             # clean up our test cookie
             if request.session.test_cookie_worked():
                 request.session.delete_test_cookie()
+
+            # redirect to self needed to call middleware again to set corrtect permission
+            redirect_to = request.GET.get("next", None)
+            if redirect_to is not None:
```

Comment:
```suggestion
            if redirect_to:
```

Reply:
I think the original version is better :D

Reply:
I'd also say the explicit None check is a bit more canonical here. 

However, @janno42's comment gave me the idea of using another value for "no redirect" than None to simplify the rest of the logic.
@fidoriel This is mainly code golf, I think we can also merge the current version.

---

The [specification for `application/x-www-form-urlencoded`](https://www.w3.org/TR/REC-html40/interact/forms.html#form-content-type) does not require a GET parameter value to be non-empty. The query string `?var=&next=&test=true` is perfectly valid.

This means that we could simplify the 4 lines here to a single line:
```python3
return redirect(reverse("evaluation:index") + "?next=" + request.GET.get("next", ""))
```

Which might append a parameter with an empty value, which we could treat just as "no value given" below in 104 with
```python3
redirect_to = request.GET.get("next", "")
if redirect_to:
    return redirect(redirect_to)
```

---

However, I think it would match our intention if we just redirect to the current url again, and not explicitly handle any GET parameters. Django has [`HttpRequest.get_full_path_info()`](https://docs.djangoproject.com/en/3.2/ref/request-response/#django.http.HttpRequest.get_full_path_info), which we could use for that:
```python3
return redirect(request.get_full_path_info())
```
You could construct cases where unconditionally keeping all GET-parameters for a second request causes bugs, but it really shouldn't happen for us as [`GET` should be a safe HTTP-method](https://www.rfc-editor.org/rfc/rfc7231#section-4.2.1).